pkg/controller/metaoperatorset/metaoperatorset_controller.go

```diff
@@ -160,41 +160,79 @@ func (r *ReconcileMetaOperatorSet) Reconcile(request reconcile.Request) (reconci
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling MetaOperatorSet")
 
+	// Fetch the MetaOperatorSet instance
+	setInstance := &operatorv1alpha1.MetaOperatorSet{}
+	if err := r.client.Get(context.TODO(), request.NamespacedName, setInstance); err != nil {
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
 	// Fetch the MetaOperatorCatalog instance
-	moc := &operatorv1alpha1.MetaOperatorCatalog{}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: "common-service"}, moc); err != nil {
+	mocList := &operatorv1alpha1.MetaOperatorCatalogList{}
+	if err := r.client.List(context.TODO(), mocList, &client.ListOptions{Namespace: setInstance.Namespace}); err != nil {
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
 		return reconcile.Result{}, err
 	}
 
+	switch mocnum := len(mocList.Items); {
+	case mocnum < 1:
+		return reconcile.Result{}, nil
+	case mocnum > 1:
+		reqLogger.Error(fmt.Errorf("multiple MetaOperatorCatalog in one namespace"),
+		"There are multiple MetaOperatorCatalog custom resource in "+
+		request.Namespace+
+		". Choose the first one "+
+		mocList.Items[0].Name+
+		" and Delete the others")
+		for i:=1; i < mocnum; i++ {
+			if err := r.client.Delete(context.TODO(), &mocList.Items[i]); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
+	moc := &mocList.Items[0]
+
 	// Initialize MetaOperatorCatalog status
 	if err := r.initOperatorStatus(moc); err != nil {
 		return reconcile.Result{}, err
 	}
 
 	// Fetch the MetaOperatorConfig instance
-	csc := &operatorv1alpha1.MetaOperatorConfig{}
-	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: "common-service"}, csc); err != nil {
+	cscList := &operatorv1alpha1.MetaOperatorConfigList{}
+	if err := r.client.List(context.TODO(), cscList, &client.ListOptions{Namespace: setInstance.Namespace}); err != nil {
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
 		return reconcile.Result{}, err
 	}
 
+	switch cscnum := len(cscList.Items); {
+	case cscnum < 1:
+		return reconcile.Result{}, nil
+	case cscnum > 1:
+		reqLogger.Error(fmt.Errorf("multiple MetaOperatorConfig in one namespace"),
+		 "There are multiple MetaOperatorConfig custom resource in "+
+		 request.Namespace+
+		 ". Choose the first one "+
+		 cscList.Items[0].Name+
+		 " and Delete the others")
+		 for i:=1; i < cscnum; i++ {
```

Comment:
Instead of deleting, can we only log a warning message and use the first one? @chenzhiwei @horis233

Reply:
Sure, I will log an error message and notify the user meta-operator to pick the first one.

Reply:
Thanks, I think we should leave the `delete` operations to end user.

Reply:
yes, we should not touch the users' data.